lib/util/http/index.ts

```diff
@@ -69,14 +74,21 @@ function applyDefaultHeaders(options: Options): void {
   };
 }
 
+// Note on types:
+// options.requestType can be either 'json' or 'buffer', but `T` should be
+// `Buffer` in the latter case.
+// We don't declare overload signatures because it's immediately wrapped by
+// `request`.
 async function gotRoutine<T>(
   url: string,
   options: GotOptions,
   requestStats: Partial<RequestStats>
 ): Promise<Response<T>> {
   logger.trace({ url, options }, 'got request');
 
-  const resp = await got<T>(url, options);
+  // Cheat the TS compiler using `as` to pick a specific overload.
+  // Otherwise it doesn't typecheck.
```

Comment:
OK. i think we can fix this later. Maybe @JamieMagee  has an idea how to solve the overloads stuff

Reply:
Playing around here, I can get the compiler to agree that `options2` is `GotJSONOptions`

```ts
const options2: Pick<
    GotOptions,
    Exclude<keyof GotOptions, keyof GotBufferOptions>
  > = options;
```